### PR TITLE
Dotted keys

### LINKED ISSUE #499: Allow keys in key-value pairs to be paths
The only remaining idea from #292 that has not been decided upon and does not have a dedicated issue.

I mean, I don't know how much I like it myself but, hey, this needs discussion so, here's a dedicated issue for it.

```
[document]
title = "Hello!"
meta.charset = "utf-8"
```

### PR DESCRIPTION
**Pull request by pradyunsg**
Closes #499

This will probably conflict with #500 since this also adds a "Keys" section.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -143,11 +159,25 @@ discouraged).
...
+In JSON land, that would give you the following structure:
+
+```json
+{
+  "name": "Orange",
+  "physical": {
+    "color": "orange",
+    "shape": "round"
+  }
+}
```
Comment (pradyunsg) [inline README.md]: I think this needs a line talking about whitespace and a section about when it's invalid.

@mojombo Something came up just now so I'm probably gonna be AWOL today so, I'm cool with you taking this into your own hands if you want. :)

**Comment by pradyunsg** [comment]
Ping @mojombo! :)

**Comment by pradyunsg** [comment]
Ping 2.0 @mojombo! 😬

**Comment by pradyunsg** [comment]
Ping 3.0!

**Comment by mojombo** [comment]
Looks great. I'm merging this as-is, but we should think about being more specific about when tables are closed, per the discussion in #499. We can continue the discussion there.

**Pull request merged**